src/core/mac/mac.cpp

```diff
@@ -750,10 +750,7 @@ void Mac::SendBeacon(Frame &aFrame)
     {
         beaconPayload->Init();
 
-        // set the Joining Permitted flag
-        GetNetif().GetIp6Filter().GetUnsecurePorts(numUnsecurePorts);
-
-        if (numUnsecurePorts)
+        if (mBeaconJoinableFlag)
```

Comment:
Joiner Router also needs to be updated to set/clear the Joining Permitted flag.

- https://github.com/openthread/openthread/blob/master/src/core/meshcop/joiner_router.cpp#L87
- https://github.com/openthread/openthread/blob/master/src/core/meshcop/joiner_router.cpp#L97